api/base/parsers.py

```diff
@@ -15,6 +17,29 @@ class JSONAPIParser(JSONParser):
     media_type = 'application/vnd.api+json'
     renderer_class = JSONAPIRenderer
 
+    def flatten_relationships(self, relationships):
+        """
+        Flattens relationships dictionary which has information needed to create related resource objects.
+
+        Validates that formatting of relationships dictionary is correct.
+        """
+        if not isinstance(relationships, dict):
+            raise ParseError()
+        related_resource = relationships.keys()[0]
```

Comment:
Since dictionaries are a hash (key order not predictable), picking the first key inherently assumes that there is only one relationship. Is this based on assumptions similar to our `target_type_` discussion earlier?